app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
+  end
+
+  def loa_no_js_fallback_link
+    if loa1_signup_context?
+      destroy_user_path
+    elsif loa3_context?
+      verify_session_path
+    end
```

Comment:
I just tested this with JS disabled, signing up via LOA3 and canceling on the OTP page deletes the users account, as expected. Is there another step I need to take to reproduce this?